Add patterns filter by post type

renatho opened the pull request:
Fixes #5187

### Changes proposed in this Pull Request

* After a deeper investigation of the mentioned issue, it seems safe to use the `current_screen` hook to register the patterns based on the current post type. It wasn't really working on some beta versions, but it was fixed after the final release with [this commit](https://github.com/WordPress/WordPress/commit/571e1e01b3f7c11c8f8d6717be8568facd7ba118) (basically, the server adds our registered patterns to the initial setup of the patterns - writing it in the HTML, and the other come later in the async fetch). Anyway, I asked for a confirmation [here](https://github.com/WordPress/gutenberg/issues/40305#issuecomment-1139962066) just to make sure I'm not missing anything.

### Testing instructions

<!--
If the functionality of this PR is behind a feature flag, please include the
following testing step, using the correct name for the feature flag. (If you
aren't sure, just ignore this step)

* Enable the feature flag: `add_filter( 'sensei_feature_flag_{THE_FLAG_NAME}', '__return_true' );`
-->
* In WP 6.0, create a new Course, and make sure only 3 patterns are listed in the patterns.
* Create a new Lesson, and make sure the other 3 patterns are listed in the patterns.
* It's nice to also test in WP 5.8 and 5.9.

renatho committed:
Filter the patterns by type

aaronfc reviewed (dismissed):
Looks good and works well! Some things I noticed with older WP versions but I think none are really important and we can "ignore" them being only minor visual things and for old WP versions.

## Using WP 5.9.3
- Some previews are displayed with an excessively long height. Example:
<img width="338" alt="image" src="https://user-images.githubusercontent.com/799065/170959609-7e0edfb8-8fe9-4dca-8f42-2f43f20221a8.png">

## Using WP 5.8.4
- Some previews are not displayed (not only for Sensei LMS but for all categories). Examples:
<img width="346" alt="image" src="https://user-images.githubusercontent.com/799065/170959955-c4b4654d-9e8d-4db6-b291-f6cf02d8bafd.png">
<img width="348" alt="image" src="https://user-images.githubusercontent.com/799065/170960027-a30e391a-7010-4ace-98b0-b3a0f1689ed7.png">
<img width="793" alt="image" src="https://user-images.githubusercontent.com/799065/170960404-004837d8-75aa-44d1-961c-7e4de8376fc3.png">

- Minor: Modal does not have a cross to close it. Clicking out of the modal or pressing escape works.
<img width="871" alt="image" src="https://user-images.githubusercontent.com/799065/170960332-7341e441-9ae9-44d8-b496-9a83f3121c1c.png">

aaronfc commented on includes/admin/class-sensei-editor-wizard.php:
```diff
@@ -98,15 +98,30 @@ public function enqueue_admin_scripts( $hook_suffix ) {
+		$post_type = $current_screen->post_type;
+
 		// Register block pattern category.
 		register_block_pattern_category(
 			self::PATTERNS_CATEGORY,
 			array( 'label' => __( 'Sensei LMS', 'sensei-lms' ) )
 		);
```
Based on [this](https://github.com/WordPress/wordpress-develop/blob/0a6c37ceb0a86bebf0b287ac12d200e67bbf948f/src/wp-includes/class-wp-block-pattern-categories-registry.php#L68) maybe we could register the category in `init` and let only blocks registration in the `current_screen` hook.

renatho replied:
Updated here: https://github.com/Automattic/sensei/pull/5202/commits/7c182f101ac23953dab55a46dad14c8d0d654b44

renatho committed:
Merge branch 'feature/course-lesson-wizard' of github.com:Automattic/sensei into add/patterns-filter-by-post-type

renatho committed:
Add WP 5.8 style fixes

renatho committed:
Move category registration to init hook

renatho commented:
Hey @aaronfc ! Thank you for your review!

> Some previews are displayed with an excessively long height.

I had noticed it and registered it as a known issue in: https://github.com/Automattic/sensei/pull/5173. I mentioned that we could wait until the final patterns to see if it will happen, and if yes, we could explore that to see what workaround we could implement. One idea I had added there is set a maximum height to the thumbnails. Does it make sense for you? To make sure we don't forget it, I'm also adding it to [a comment in this issue](https://github.com/Automattic/sensei/issues/5136#issuecomment-1141230360).

> Some previews are not displayed (not only for Sensei LMS but for all categories)

I thought it had gone away. 😞 Reopened [this issue](https://github.com/Automattic/sensei/issues/5165).

> Modal does not have a cross to close it.

Good catch! Fixed in https://github.com/Automattic/sensei/pull/5202/commits/f22a70af8610e2a8eea61ab2248d0896a5da0a30

aaronfc reviewed (approved):
Everything clear!

Looks good and works well! 🚀
